control/metrics.go

```diff
@@ -453,6 +453,14 @@ func (mc *metricCatalog) RmUnloadedPluginMetrics(lp *loadedPlugin) {
 	mc.mutex.Lock()
 	defer mc.mutex.Unlock()
 	mc.tree.DeleteByPlugin(lp)
+
+	// Update metric catalog keys
+	mc.keys = []string{}
+	mts := mc.tree.gatherMetricTypes()
+	for _, m := range mts {
+		mc.keys = append(mc.keys, m.Namespace().Key())
+	}
```

Comment:
It's interesting, I noticed just yesterday that unloading a running plugin does not kill it.  I was going to investigate this more tomorrow as I do not recall if the original intent was for the plugin to continue to run but not